src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -0,0 +1,54 @@
+package io.supertokens.totp;
+
+import java.io.IOException;
+
+import io.supertokens.Main;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.sqlStorage.TOTPSQLStorage;
+import io.supertokens.storageLayer.StorageLayer;
+
+public class Totp {
+
+    public static CreateDeviceResponse createDevice(Main main, String userId, String deviceName, int skew, int period)
+            throws IOException {
+
+        TOTPSQLStorage totpStorage = StorageLayer.getTOTPStorage(main);
+        String secret = GenerateDeviceSecret.generate();
+
+        if (userId == null || deviceName == null || secret == null) {
+            throw new IllegalArgumentException("userId, deviceName and secret cannot be null");
+        }
```

Comment:
in put validation should not happen here. It should happen in the API layer. remove this

Reply:
Done.